codalab/server/bundle_manager.py

```diff
@@ -707,8 +722,14 @@ def _get_staged_bundles_to_run(self, workers, user_info_cache):
             failures.append(
                 self._check_resource_failure(
                     bundle_resources.disk,
-                    user_fail_string='Requested more disk (%s) than user disk quota left (%s)',
-                    user_max=self._model.get_user_disk_quota_left(bundle.owner_id, user_info),
+                    user_fail_string='Requested more disk (%s) than user disk quota left (%s) by %s',
+                    # Return the larger number between 1.0 and disk quota after deducting
+                    # DISK_QUOTA_SLACK_BYTES to avoid float overflow
```

Comment:
How does max'ing with 1 deal with float overflow?  Shouldn't be necessary?

Reply:
After deducting 0.5G from the original max disk quota (e.g. 0.1G - 0.5G), the result could become a negative value, which is probably not what we want when setting the `user_max` value.

Reply:
If the value is negative, then the job will fail anyway...  I'd just remove the max if the behavior is correct to keep things simple.